pkg/subctl/lighthouse/dns/ensure.go

```diff
@@ -227,28 +228,31 @@ func setupOpenShift(status *cli.Status, clientSet *clientset.Clientset, repo str
 		// Assume we're not on OpenShift
 		return nil
 	}
-	deployment, err := deployments.Get("dns-operator", metav1.GetOptions{})
-	if err != nil {
-		return err
-	}
-	if deployment == nil {
-		// Assume we're not on OpenShift
-		return nil
-	}
-	for i, container := range deployment.Spec.Template.Spec.Containers {
-		if container.Name == "dns-operator" {
-			deployment.Spec.Template.Spec.Containers[i].Image = repo + operatorImage + ":" + version
-			for j, env := range container.Env {
-				if env.Name == "IMAGE" {
-					deployment.Spec.Template.Spec.Containers[i].Env[j].Value = repo + openShiftCoreDNSImage + ":" + version
+	retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		deployment, err := deployments.Get("dns-operator", metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
```

Comment:
This should/will return `errors.IsNotFound` if it doesn't exist so I think we should check for that and return nil.  Also there's no need for the nil check on L236 - it won't return nil if no error is returned (in fact it won't return nil either way).

On a side note, there's also no need for the nil check on L227 above as that call never returns nil.

Reply:
Done, thanks!